pkg/splunk/enterprise/configuration.go

```diff
@@ -308,11 +306,41 @@ func validateCommonSplunkSpec(spec *enterpriseApi.CommonSplunkSpec) error {
 		return fmt.Errorf("Negative value (%d) is not allowed for Readiness probe intial delay", spec.ReadinessInitialDelaySeconds)
 	}
 
+	// if not provided, set default values for imagePullSecrets
+	err := ValidateImagePullSecrets(ctx, c, cr, spec)
+	if err != nil {
+		return err
+	}
+
 	setVolumeDefaults(spec)
 
 	return ValidateSpec(&spec.Spec, defaultResources)
 }
 
+// ValidateImagePullSecrets sets default values for imagePullSecrets if not provided
+func ValidateImagePullSecrets(ctx context.Context, c splcommon.ControllerClient, cr splcommon.MetaObject, spec *enterpriseApi.CommonSplunkSpec) error {
```

Comment:
Could we extract the spec from the cr here rather than passing it all the way from validateCommonSplunkSpec?

Reply:
The CR object here is of type [splcommon.MetaObject](https://github.com/splunk/splunk-operator/blob/93edb2ca16aba6bbab1ef3accd54532416db812f/pkg/splunk/common/types.go#L111) and the enterpriseApi.CommonSplunkSpec cannot be extracted from the same. The reason we have the common metaObject is to make sure that we are able to call into a common function for all CRs regardless of the type of CR.